cranelift/codegen/src/ir/immediates.rs

```diff
@@ -800,6 +800,26 @@ impl Ieee32 {
     pub fn is_zero(&self) -> bool {
         self.as_f32() == 0.0
     }
+
+    /// Returns the smallest integer greater than or equal to `self`.
+    pub fn ceil(self) -> Self {
+        Self::with_float(self.as_f32().ceil())
+    }
+
+    /// Returns the largest integer less than or equal to `self`.
+    pub fn floor(self) -> Self {
+        Self::with_float(self.as_f32().floor())
+    }
+
+    /// Returns the integer part of `self`. This means that non-integer numbers are always truncated towards zero.
+    pub fn trunc(self) -> Self {
+        Self::with_float(self.as_f32().trunc())
+    }
+
+    /// Returns the nearest integer to `self`. Round half-way cases away from `0.0`.
+    pub fn nearest(self) -> Self {
+        Self::with_float(self.as_f32().round())
```

Comment:
`round` is not the same as `nearest`. I think I once implemented the correct nearest in the past: https://github.com/bytecodealliance/wasmtime/pull/2171

Reply:
Ah, sorry I missed this... @afonso360 could you look into this further?

Reply:
Rust has not had this operation for very many years. Here it seems someone finally want to add it:
https://github.com/rust-lang/rust/pull/95317

Reply:
Oh wow! Great catch!

Is there a simple way to organize the code so that we can reuse that implementation?

I'm not sure we can use nightly features in wasmtime/cranelift.

Otherwise we can probably duplicate the function pending stabilization of the rust implementation.